modules/charon-utils/src/main/java/org/wso2/charon3/utils/usermanager/InMemoryUserManager.java

```diff
@@ -41,16 +40,14 @@
 /**
  * This is a sample dynamic user store.
  */
-public class InMemoryUserManager implements UserManager {
-    private static final Logger logger = LoggerFactory.getLogger(InMemoryUserManager.class);
+public class InMemoryUserManager implements ResourceHandler<User> {
+
     //in memory user manager stores users
     ConcurrentHashMap<String, User> inMemoryUserList = new ConcurrentHashMap<String, User>();
-    ConcurrentHashMap<String, Group> inMemoryGroupList = new ConcurrentHashMap<String, Group>();
 
 
     @Override
-    public User createUser(User user, Map<String, Boolean> map)
-            throws CharonException, ConflictException, BadRequestException {
+    public User create(User user, Map<String, Boolean> map) throws AbstractCharonException {
```

Comment:
It is not recommended to change public method signatures as they can be used from some other places as well. We need to preserve the backward compatibility.